# To Foul or Not to Foul - Solution


# The Problem

### Is it smart to foul at the end of the basketball game?

In the 2005 Conference USA basketball tournament, Memphis was two points behind Louisville. At the buzzer, Memphis’s Darius Washington attempted a 3-pointer. He missed but was fouled so he went to the line for three free throws. Each made free throw is worth 1 point.



## Exercise 1

Darius Washington was a 72% free-throw shooter. Write a Monte Carlo simulation to find the probability that Memphis will win, lose or go into overtime. Note that in each "experiment" (i.e. iteration of a loop), you have to do three free-throws.

In [1]:
import numpy as np
import matplotlib.pyplot as plt
from matplotlib import colors
import random as rand

In [8]:
N = 1000000
FTpercentage = 0.72

Nwin = 0
Nlose = 0
Novertime = 0

for i in range(N):
    Nmade = 0
    s1 = rand.random()
    if s1 < FTpercentage:
        Nmade = Nmade + 1

    s1 = rand.random()
    if s1 < FTpercentage:
        Nmade = Nmade + 1

    s1 = rand.random()
    if s1 < FTpercentage:
        Nmade = Nmade + 1

    if Nmade == 3:
        Nwin = Nwin + 1
    elif Nmade == 2:
        Novertime = Novertime + 1
    else:
        Nlose = Nlose + 1

pwin = Nwin/N*100
plose = Nlose/N*100
povertime = Novertime/N*100

print("prob of Memphis winning in regulation = ", pwin)
print("prob of overtime = ", povertime)
print("prob of Memphis losing in regulation = ", plose)

prob of Memphis winning =  37.383
prob of overtime =  43.5154
prob of Memphis losing =  19.101599999999998


## Exercise 2

If Washington had not been fouled, he may have made the 3-pointer. If he is a 40% 3-point shooter, do you think Louisville was smart to foul? Why or why not?

The probability of Lousiville losing in regulation by a 3 point shot is 40%. The probability of Louisville losing *in regulation* after fouling is 37.4%. Since they are not guaranteed to win in overtime, the overall probality of losing will surely be greter than 37%. So fouling is probability a bad idea. But we don't know exactly how much greater unless we make some assumptions and simulate it.

Let's assume that the probability of winning a game for each team in overtime is 50%. (After all, the two teams seem to be closely matched.)

In [11]:
N = 1000000
FTpercentage = 0.72

NwinFT = 0
NloseFT = 0
Novertime = 0

NwinOT = 0
NloseOT = 0

for i in range(N):
    Nmade = 0
    s1 = rand.random()
    if s1 < FTpercentage:
        Nmade = Nmade + 1

    s1 = rand.random()
    if s1 < FTpercentage:
        Nmade = Nmade + 1

    s1 = rand.random()
    if s1 < FTpercentage:
        Nmade = Nmade + 1

    if Nmade == 3:
        NwinFT = NwinFT + 1
    elif Nmade == 2:
        Novertime = Novertime + 1
        r = rand.random()
        if r < 0.5:
            NwinOT = NwinOT + 1
        else:
            NloseOT = NloseOT + 1
    else:
        NloseFT = NloseFT + 1

pwinFT = NwinFT/N*100
ploseFT = NloseFT/N*100
povertime = Novertime/N*100

Nwin = NwinFT + NwinOT
Nlose = NloseFT + NloseOT
pwin = Nwin/N * 100
plose = Nlose/N * 100

print("prob of Memphis winning in regulation = ", pwinFT)
print("prob of overtime = ", povertime)
print("prob of Memphis losing in regulation = ", ploseFT)

print("prob of Memphis winning = ", pwin)
print("prob of Memphis losing = ", plose)

prob of Memphis winning in regulation =  37.2952
prob of overtime =  43.5412
prob of Memphis losing in regulation =  19.1636
prob of Memphis winning =  59.054700000000004
prob of Memphis losing =  40.9453


For Louisville, their probability of winning if Memphis shoots a 3 point shot is 60%. If Memphis shoots FTs, Louisville has a probability of winning of 41%.

**Louisville should not have fouled.**